File: engine/src/main/java/com/terralite/engine/world/WorldAccess.java
package com.terralite.engine.world;

import com.terralite.engine.chunk.Chunk;
import com.terralite.engine.chunk.ChunkPos;

import java.util.Collection;
import java.util.Optional;

public interface WorldAccess {
    Optional<Chunk> getChunk(ChunkPos pos);

    Chunk requireChunk(ChunkPos pos);

    boolean containsChunk(ChunkPos pos);

    Collection<ChunkPos> chunkPositions();
}

File: engine/src/commonMain/kotlin/com/terralite/engine/world/WorldAccess.kt
package com.terralite.engine.world

import com.terralite.engine.chunk.Chunk
import com.terralite.engine.chunk.ChunkPos

interface WorldAccess {
    fun getChunk(pos: ChunkPos): Chunk?
    fun requireChunk(pos: ChunkPos): Chunk
    fun containsChunk(pos: ChunkPos): Boolean
    fun chunkPositions(): Collection<ChunkPos>
}
